app/src/main/java/com/nextcloud/talk/diagnose/DiagnoseContentComposable.kt

```diff
@@ -59,19 +115,84 @@ fun DiagnoseContentComposable(data: State<List<DiagnoseActivity.DiagnoseElement>
                 }
             }
         }
+        ShowNotificationData(isLoading.value, showDialog.value, context, message.value, diagnoseViewModel)
     }
 }
 
-@Preview(showBackground = true)
 @Composable
-fun DiagnoseContentPreview() {
-    val state = remember {
-        mutableStateOf(
-            listOf(
-                DiagnoseActivity.DiagnoseElement.DiagnoseHeadline("Headline"),
-                DiagnoseActivity.DiagnoseElement.DiagnoseEntry("Key", "Value")
+@Suppress("LongMethod")
+fun ShowNotificationData(
+    isLoading: Boolean,
+    showDialog: Boolean,
+    context: Context,
+    message: String,
+    diagnoseViewModel: DiagnoseViewModel
+) {
+    if (isLoading) {
+        Box(
+            modifier = Modifier.fillMaxSize(),
+            contentAlignment = Alignment.Center
+        ) {
+            CircularProgressIndicator()
+        }
+    }
+    if (showDialog) {
+        Dialog(
+            onDismissRequest = { diagnoseViewModel.dismissDialog() },
+            properties = DialogProperties(
+                dismissOnClickOutside = true,
+                usePlatformDefaultWidth = false
             )
-        )
+        ) {
+            Surface(
+                shape = MaterialTheme.shapes.medium,
+                tonalElevation = 8.dp,
+                modifier = Modifier
+                    .wrapContentSize()
+                    .padding(16.dp)
+            ) {
+                Column(modifier = Modifier.padding(16.dp)) {
+                    Text(
+                        text = stringResource(R.string.nc_test_results),
+                        style = MaterialTheme.typography
+                            .titleMedium
+                    )
```

Comment:
I think in the dialog it would make sense to add an explanation to the user that a notification with the text "Testing push notifications" should pop up. 
Otherwise some users might think that "Push notification sent successfully" means that everything is working (which is only true for the server side), even if the notification did not make it to the phone.